### PR TITLE
Add acceptance tests for how provider handles `scopes` argument

### PR DESCRIPTION
**Pull request by SarahFrench**
This PR adds acceptance tests for usage of scopes that demonstrate:

- how the provider behaves when provider configuration arguments come from different sources ( config vs ENVs)
    - There are no ENVs for scopes
    - There is a default set of scopes used if the argument isn't supplied
- schema-level validation that's in place
    - There isn't any validation about empty lists, instead the empty list is ignored
- use cases: how does this argument impact the providers behaviour in plan/apply
    - I've implemented a test case where setting the scopes to incorrect values causes provisioning to fail.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by SarahFrench** [comment]
TeamCity tests:

- [SDK](https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_BETA_MMUPSTREAMTESTS_GOOGLEBETA_PACKAGE_PROVIDER/240887?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true)
- [PF](https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_BETA_MMUPSTREAMTESTS_GOOGLEBETA_PACKAGE_FWPROVIDER/240888?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true)

**Comment by SarahFrench** [comment]
I'm not sure what's going on in the failure above, very hard to search in the logs

File: mmv1/third_party/terraform/provider/provider_scopes_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+// It is a counterpart to TestAccFwProvider_scopes
+func TestAccSdkProvider_scopes(t *testing.T) {
+	testCases := map[string]func(t *testing.T){
+		// Configuring the provider using inputs
+		"default scopes are used when there are no user inputs": testAccSdkProvider_scopes_providerDefault,
+		"scopes can be set in config":                           testAccSdkProvider_scopes_setInConfig,
+		//no ENVs to test
+
+		// Schema-level validation
+		"when scopes is set to an empty array in the config the value is ignored and default scopes are used": testAccSdkProvider_scopes_emptyArray,
```
Comment (c2thorn) [inline mmv1/third_party/terraform/provider/provider_scopes_test.go:24]: nice capturing this case!

**Review by c2thorn — APPROVED** [review]

File: mmv1/third_party/terraform/provider/provider_scopes_test.go (hunk 1 above)
Comment (SarahFrench) [reply mmv1/third_party/terraform/provider/provider_scopes_test.go:24]: Thanks 😁 We could add some validation in future to make the scenario more explicit, i.e. block empty arrays so users aren't lulled into a false sense of using no scopes, but that'd be a breaking change. And I don't even know if setting zero scopes is something users would attempt to do, but I can imagine there's someone out there doing it!

**Pull request merged**